File: python/ex_02.py
# Define a function max() that takes two numbers as arguments and returns
# the largest of them.


def maxNum(inputNum1, inputNum2):
    """Find the largest of two input numbers"""
    if inputNum1 > inputNum2:
        return inputNum1
    else:
        return inputNum2


def maxOfThree(inputNum1, inputNum2, inputNum3):
    """Find the largest of three input numbers"""
    firstMax = maxNum(inputNum1, inputNum2)
    return maxNum(firstMax, inputNum3)


print("Please enter a number:")
userNum1 = input()
print("Please enter a second number:")
userNum2 = input()
print("Please enter a third number:")
userNum3 = input()
print("The largest of the three is " \
      + maxOfThree(userNum1, userNum2, userNum3))

File: c/ex_02.c
/* Define a function max_of_three() that takes three numbers as 
 * arguments and returns the largest of them. */

#include <stdio.h>

int 
max(int num1, int num2)
{
    if (num1 > num2)
    {
        return num1;
    }
    else
    {
        return num2;
    }
}

int 
max_of_three(int num1, int num2, int num3)
{
    int first_max = max(num1, num2);

    if (first_max > num3)
    {
        return first_max;
    }
    else
    {
        return num3;
    }
}

int 
main(void)
{
    int num1, num2, num3;

    printf("Please enter 3 numbers: ");
    scanf(" %d %d %d", &num1, &num2, &num3);

    printf("The largest of the three is %d\n", 
            max_of_three(num1, num2, num3));

    return 0;
}


